src/System.Linq.Expressions/src/System/Linq/Expressions/CatchBlock.cs

```diff
@@ -158,7 +157,20 @@ public static CatchBlock MakeCatchBlock(Type type, ParameterExpression variable,
         {
             ContractUtils.RequiresNotNull(type, nameof(type));
             ContractUtils.Requires(variable == null || TypeUtils.AreEquivalent(variable.Type, type), nameof(variable));
-            if (variable != null && variable.IsByRef)
+            if (variable == null)
+            {
+                TypeUtils.ValidateType(type, nameof(type));
```

Comment:
Might be a good idea, although need to be sure that IsRef is not going to make problems for byref parameters or returns.